### PR TITLE
 🗂 Organize all the components!

### PR DESCRIPTION
**Pull request by andycochran**
This PR, in individual commits, will… 

- [x] Move auth components to `auth/` subdirectory
- [x] Move breadcrumb components to `ui/` subdirectory
- [x] Rename `confirmation-message` to `confirmation-modal` (It renders modal and buttons. The "message" is passed to it.)
- [x] Move `page-nav` component to `ui/` subdirectory
- [x] Move `assistance-message.hbs` to `messages/assistance.hbs` (This assumes we will have other repeated messages.)
- [x] Move `character-counter` component to `saveable-form/` subdirectory
- [x] Nix `validation-message` component, and use proper classes that follow our style guide (The component unnecessarily adds files and only invokes a span with a class in 1 place.)
- [x] Move `form-validator` component to `saveable-form/validator` (This moves and renames it, which make the template easier to read. But wanna make sure folks agree it's easier to read.)
- [x] Move `save-button` component from `packages/` to `saveable-form/` subdirectory
- [x] Move `land-use-action` component to `packages/` subdirectory
- [x] Move `project-geography` component to `packages/` subdirectory
- [x] Move `proposed-development-types-list` component to `packages/` subdirectory
- [x] Move `saveable-form` component to subdirectory as `saveable-form/index`

Addresses #329.

### CODE REVIEW

File: client/app/components/packages/pas-form/project-geography.hbs [hunk 1]
```diff
@@ -10,7 +10,7 @@
     </p>
 
     <div data-test-section="project-geography">
-      <ProjectGeography
+      <Packages::ProjectGeography
```
Comment (trbmcginnis) [inline client/app/components/packages/pas-form/project-geography.hbs:13]: Just leaving a note for the future. We may want to put this component in `<Packages::PasForm::ProjectGeography>`

File: client/app/components/packages/pas-form/proposed-land-use-actions.hbs [hunk 2]
```diff
@@ -8,7 +8,7 @@
       Note: This is preliminary information and may change following discussion with NYC Planning.
     </p>
     <div data-test-section="land-use-actions">
-      <LandUseAction
+      <Packages::LandUseAction
```
Comment (trbmcginnis) [inline client/app/components/packages/pas-form/proposed-land-use-actions.hbs:11]: Depending on future packages/whether this component is reusable, could eventually move to `<Packages::PasForm::LandUseAction>`

File: client/app/components/packages/pas-form/show.hbs [hunk 3]
```diff
@@ -465,7 +465,7 @@
         </A.Prompt>
         <A.Field>
           <span data-test-proposedDevelopmentTypes>
-            <ProposedDevelopmentTypesList @pasForm={{@package.pasForm}} />
+            <Packages::ProposedDevelopmentTypesList @pasForm={{@package.pasForm}} />
```
Comment (trbmcginnis) [inline client/app/components/packages/pas-form/show.hbs:468]: Could this go in `<Packages::PasForm::ProposedDevelopmentTypesList>` <-- this seems quite unique, not sure if we'd reuse this

**Comment by godfreyyeung** [comment]
> * [x]  Nix `validation-message` component, and use proper classes that follow our style guide (The component unnecessarily adds files and only invokes a span with a class in 1 place.)

Sorry I missed this discussion. Can you explain this a bit more and the reasons behind this?

**Comment by allthesignals** [comment]
Out of curiosity, any idea why it lost the commit history after a `git mv`? —> https://github.com/NYCPlanning/labs-applicant-portal/blob/develop/client/app/components/saveable-form/validator.js

**Pull request merged**